api/src/main/proto/stellarstation/api/v1/groundstation/groundstation.proto

```diff
@@ -265,6 +266,13 @@ message GroundStationStreamRequest {
     // - `plan_monitoring_event.ground_station_event` must be sent once at each of the transitions
     //   documented in `PlanLifecycleEvent` when executing a plan.
     StreamEvent stream_event = 4;
+
+    // An event generated by the client during a plan that might affect the plan's execution. The
+    // event is consumed by the server and will not be directly forwarded to the satellite operator.
+    //
+    // This field is ignored unless the client is connected to an endpoint at the edge device
+    // (StarPass).
```

Comment:
For clients that are not connected at the edge device, wouldn't they expect similar functionality?

Reply:
This is a tough question. It would help to brainstorm about what the server would do when it receives a plan abortion event.
- Stop processes and destroy resources created for plan execution. [Only at the edge device]
- Move the plan to a failed status.
- Logs the failure and the reason.

The first two actions will be taken whether the plan completes successfully or fails prematurely. The event only helps the actions to occur early. I thought it was worth doing but I'm now not very sure if it would be worth the complexity. Maybe we just need logging?

Reply:
Let me take a step back.

Inside StreamEvent there is already another message called PlanLifeCycleEvent. Will that still be sent to the server in the new system? If yes, then this new PlanClientEvent can be edge device only.

Related to the above: since you are also adding a message called PlanLifeCycleEvent in this PR it might be confusing since there is already another message with the same name.

Reply:
Good catch :)

`monitoring.PlanLifecycleEvent` is generated by StellarStation and consumed by satellite operators. I actually don't know whether we should require the client of the groundstation API to generate it. At least within scope of this PR, it is StarPass to generate them. However I can't really decide if this will change or not.

It is only a matter of naming. I will take more time to look for better names.